src/workflows-app/components/WorkflowInfoBox.ts

```diff
@@ -0,0 +1,147 @@
+import { useCallback, useState } from 'react';
+import { div, h, span } from 'react-hyperscript-helpers';
+import { Link } from 'src/components/common';
+import { icon } from 'src/components/icons';
+import { collapseStatus, makeStatusLine } from 'src/components/job-common';
+import { useCancellation, useOnMount } from 'src/libs/react-utils';
+import { makeCompleteDate } from 'src/libs/utils';
+import ViewWorkflowScriptModal from 'src/workflows-app/components/ViewWorkflowScriptModal';
+import { fetchMetadata } from 'src/workflows-app/utils/cromwell-metadata-utils';
+
+import { loadAppUrls } from '../utils/app-utils';
+import { TroubleshootingBox } from './TroubleshootingBox';
+
+export type WorkflowInfoBoxProps = {
+  name: string;
+  namespace: string;
+  submissionId: string;
+  workflowId: string;
+  workspaceId: string;
+  showLogModal: (modalTitle: string, logsArray: string, tesLog: string) => void;
+};
+
+type FetchedWorkflowInfoData = {
+  start: string;
+  end?: string;
+  wdlScript?: string;
+  status: string;
+  workflowLog: string;
+  executionDirectory: string;
+};
+
+export const WorkflowInfoBox: React.FC<WorkflowInfoBoxProps> = (props) => {
+  const signal = useCancellation();
+  const [workflowInfo, setWorkflowInfo] = useState<FetchedWorkflowInfoData | undefined>(undefined);
```

Comment:
Another stylistic choice... FYI, React's types have a shorthand for this:

```suggestion
  const [workflowInfo, setWorkflowInfo] = useState<FetchedWorkflowInfoData>();
```